### ISSUE TITLE
Can't mock `electron` api

### ISSUE THREAD

**Issue by cawa-93** [vitest-dev/vitest#425]
### Describe the bug

I want to test the algorithm for creating a window in the electron. But for some reason I can't create mock. Calling the `vi.fn()` in the `vi.mock` leads to an error

### Reproduction

[Reproduction](https://github.com/cawa-93/vite-electron-builder/blob/vitest/packages/main/tests/unit.spec.ts)
```ts
// tests/unit.spec.ts 
import * as electron from 'electron';
import {expect, test, vi} from 'vitest';

vi.mock('electron', () => {
  return {
    BrowserWindow: vi.fn(),
  };
});

test('window', async () => {
  expect(vi.isMockFunction(electron.BrowserWindow)).toEqual(true);
});
```

```node
 FAIL  tests/unit.spec.ts [ tests/unit.spec.ts ]
ReferenceError: Cannot access '__vite_ssr_import_1__' before initialization
 ❯ tests/unit.spec.ts:3:20
      2| import {expect, test, vi} from 'vitest';
      3| 
      4| vi.mock('electron', () => {
       |                   ^
      5|   return {
      6|     BrowserWindow: vi.fn(),
 ❯ tests/unit.spec.ts:6:37
 ❯ async collectTests ../../node_modules/vitest/dist/entry.js:1363:7
 ❯ async startTests ../../node_modules/vitest/dist/entry.js:1569:17
 ❯ async ../../node_modules/vitest/dist/entry.js:1596:7
 ❯ async withEnv ../../node_modules/vitest/dist/entry.js:1281:5
```

You can setup reproduction by follow:
```bash
git clone -b vitest https://github.com/cawa-93/vite-electron-builder.git
npm i
npm run test:main
```

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 AMD Ryzen 7 4800H with Radeon Graphics
    Memory: 7.31 GB / 15.42 GB
  Binaries:
    Node: 16.13.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (96.0.1054.62)
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    @vitejs/plugin-vue: 2.0.1 => 2.0.1
    vite: 2.7.10 => 2.7.10
    vitest: 0.0.129 => 0.0.129
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
The bug is that `vi.mock` is getting hoisted, but import from `'vitest'` isn't.

Until the bug is fixed, you can use `--global` flag to not import it.

**Comment by sheremet-va** [comment]
Fixed in [v0.0.132](https://github.com/vitest-dev/vitest/releases/tag/v0.0.132) by https://github.com/vitest-dev/vitest/commit/6fe819a
